pkg/cfn/builder/vpc.go

```diff
@@ -320,3 +322,83 @@ func (n *NodeGroupResourceSet) addResourcesForSecurityGroups() {
 		}
 	}
 }
+
+func (c *ClusterResourceSet) haNAT() {
+
+	for _, az := range c.spec.AvailabilityZones {
+		alphanumericUpperAZ := strings.ToUpper(strings.Join(strings.Split(az, "-"), ""))
+
+		// Allocate an EIP
+		c.newResource(fmt.Sprintf("%v%v", "NATIP", alphanumericUpperAZ), &gfn.AWSEC2EIP{
+			Domain: gfn.NewString("vpc"),
+		})
+		// Allocate a NAT gateway in the public subnet
+		refNG := c.newResource(fmt.Sprintf("%v%v", "NATGateway", alphanumericUpperAZ), &gfn.AWSEC2NatGateway{
+			AllocationId: gfn.MakeFnGetAttString(fmt.Sprintf("%v%v%v", "NATIP", alphanumericUpperAZ, ".AllocationId")),
+			SubnetId:     gfn.MakeRef(fmt.Sprintf("%v%v", "SubnetPublic", alphanumericUpperAZ)),
+		})
+
+		// Allocate a routing table for the private subnet
+		refRT := c.newResource(fmt.Sprintf("%v%v", "PrivateRouteTable", alphanumericUpperAZ), &gfn.AWSEC2RouteTable{
+			VpcId: c.vpc,
+		})
+		// Create a route that sends Internet traffic through the NAT gateway
+		c.newResource(fmt.Sprintf("%v%v", "NATPrivateSubnetRoute", alphanumericUpperAZ), &gfn.AWSEC2Route{
+			RouteTableId:         refRT,
+			DestinationCidrBlock: internetCIDR,
+			NatGatewayId:         refNG,
+		})
+		// Associate the routing table with the subnet
+		c.newResource("RouteTableAssociationPrivate"+alphanumericUpperAZ, &gfn.AWSEC2SubnetRouteTableAssociation{
```

Comment:
Why use sprintf in line 346 and not here?

Reply:
A bit rusty I guess. I will update everything to string concatenation